File: src/main/java/com/itechart/demojavakotlin/entity/UserEntity.java
package com.itechart.demojavakotlin.entity;

import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "users")
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", insertable = false, updatable = false)
    private UUID id;

    @Column(name = "money")
    private BigDecimal money;

    @Column(name = "name", nullable = false)
    private String name;

    @OneToMany(mappedBy = "userId", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<TicketEntity> tickets;
}

File: kotlinpath/demojavakotlin/src/main/kotlin/by/itechart/demojavakotlin/entity/UserEntity.kt
package by.itechart.demojavakotlin.entity

import java.math.BigDecimal
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "users")
data class UserEntity(
        @Id
        @Column(name = "id", insertable = false, updatable = false)
        @GeneratedValue
        var id: UUID = UUID.randomUUID(),

        @Column(name = "money")
        val money: BigDecimal? = BigDecimal(0),

        @Column(name = "name", nullable = false)
        val name: String,

        @OneToMany(mappedBy = "userId", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
        val tickets: List<TicketEntity>? = null
)
